pkg/common/oras/authprovider/azure/azureidentity.go

```diff
@@ -29,14 +29,17 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
-	"github.com/Azure/azure-sdk-for-go/services/preview/containerregistry/runtime/2019-08-15-preview/containerregistry"
+	"github.com/Azure/azure-sdk-for-go/sdk/containers/azcontainerregistry"
 )
 
 type azureManagedIdentityProviderFactory struct{}
-type azureManagedIdentityAuthProvider struct {
-	identityToken azcore.AccessToken
-	clientID      string
-	tenantID      string
+type MIAuthProvider struct {
+	identityToken           azcore.AccessToken
+	clientID                string
+	tenantID                string
+	authClientFactory       func(serverURL string, options *azcontainerregistry.AuthenticationClientOptions) (AuthClient, error)
```

Comment:
curious, have you considered creating a separate interface for these 3 methods and then having the `MIAuthProvider` type have the new interface instance as a new field? That is essentially the same as what you are doing now except as a separate type. Any preferences @susanshi @duffney @binbin-li ?

Reply:
I see, if the token properties are different type to the grouping of the methods, we should keep them separate.

Reply:
seems it's mainly for making unit test easier. I would prefer defining interfaces for those methods.